pyccel/pyccel
# Speed up unit tests by running them in parallel with pytest-xdist

Install pytest-xdist and use it to run unit tests with as many processes as possible. Fixes #748 

**Commit Summary**
- Install pytest-xdist
- Run tests with option `-n auto`
- Add `pyccel_clean.py` file
- Add `pyccel-clean` script
- Only copy and recompile internal libraries if they are not already present
- Ensure that all generated files contain a tag (this was not the case when translating modules)
- Increase the size of the random string (seed is not changed as it has no effect for `random.SystemRandom()`)
- Remove duplicate function `random_string` from `parser/utilities.py` where it was unused
- Only teardown tests when running in serial
- Run parallel tests in a separate section
- invalidate caches before importing newly created file (this was done for the serial case, and is added for mpi cases). Fixes #811 
- Add `xdist_incompatible` pytest marker. This is necessary as some pyccel tests reuse files. pyccel does not mangle names in order to allow importing so this causes name clashes when running in parallel
- Remove tests/printers folder. This contains tests for printing sympy objects with the pyccel printers which no longer makes any sense. Occasionally pytest tries to parse these files despite the `--ignore` flag and fails
